### ISSUE TITLE
[ONNX] The setting of 'dynamic_shapes' in Dynamo Export is not taking effect.

### ISSUE THREAD

**Issue by cclinet** [pytorch/pytorch#119180]
### 🐛 Describe the bug

I compile the code from https://pytorch.org/docs/stable/onnx_dynamo.html#torch.onnx.dynamo_export
like below

```python
import torch

class MyModel(torch.nn.Module):
    def __init__(self) -> None:
        super().__init__()
        self.linear = torch.nn.Linear(2, 2)
    def forward(self, x, bias=None):
        out = self.linear(x)
        out = out + bias
        return out
model = MyModel()
kwargs = {"bias": 3.}
args = (torch.randn(2, 2, 2),)

export_options = torch.onnx.ExportOptions(dynamic_shapes=True)
onnx_program = torch.onnx.dynamo_export(
    model,
    *args,
    **kwargs,
    export_options=export_options)
print(onnx_program.model_proto.graph.input[0])
```

Then I got the result
```
name: "l_x_"
type {
  tensor_type {
    elem_type: 1
    shape {
      dim {
        dim_param: "2"
      }
      dim {
        dim_param: "2"
      }
      dim {
        dim_param: "2"
      }
    }
  }
}

```

### Versions
```
Collecting environment information...
PyTorch version: 2.2.0+cu121
Is debug build: False
CUDA used to build PyTorch: 12.1
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: Could not collect
CMake version: version 3.26.3
Libc version: glibc-2.31

Python version: 3.10.13 (main, Sep 11 2023, 13:44:35) [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-5.4.0-132-generic-x86_64-with-glibc2.31
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: 
GPU 0: Tesla V100-SXM2-32GB
GPU 1: Tesla V100-SXM2-32GB

Nvidia driver version: 525.147.05
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
Address sizes:                   46 bits physical, 48 bits virtual
CPU(s):                          20
On-line CPU(s) list:             0-19
Thread(s) per core:              1
Core(s) per socket:              20
Socket(s):                       1
NUMA node(s):                    1
Vendor ID:                       GenuineIntel
CPU family:                      6
Model:                           85
Model name:                      Intel(R) Xeon(R) Platinum 8255C CPU @ 2.50GHz
Stepping:                        5
CPU MHz:                         2499.998
BogoMIPS:                        4999.99
Hypervisor vendor:               KVM
Virtualization type:             full
L1d cache:                       640 KiB
L1i cache:                       640 KiB
L2 cache:                        80 MiB
L3 cache:                        35.8 MiB
NUMA node0 CPU(s):               0-19
Vulnerability Itlb multihit:     KVM: Vulnerable
Vulnerability L1tf:              Mitigation; PTE Inversion
Vulnerability Mds:               Vulnerable: Clear CPU buffers attempted, no microcode; SMT Host state unknown
Vulnerability Meltdown:          Mitigation; PTI
Vulnerability Mmio stale data:   Vulnerable: Clear CPU buffers attempted, no microcode; SMT Host state unknown
Vulnerability Retbleed:          Vulnerable
Vulnerability Spec store bypass: Vulnerable
Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; Retpolines, STIBP disabled, RSB filling, PBRSB-eIBRS Not affected
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Vulnerable: Clear CPU buffers attempted, no microcode; SMT Host state unknown
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl cpuid tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_single pti fsgsbase bmi1 hle avx2 smep bmi2 erms invpcid rtm mpx avx512f avx512dq rdseed adx smap clflushopt clwb avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 arat avx512_vnni

Versions of relevant libraries:
[pip3] numpy==1.26.3
[pip3] onnx==1.15.0
[pip3] onnxscript==0.1.0.dev20240205
[pip3] torch==2.2.0
[pip3] triton==2.2.0
[conda] numpy                     1.26.3                   pypi_0    pypi
[conda] torch                     2.2.0                    pypi_0    pypi
[conda] triton                    2.2.0                    pypi_0    pypi
```

**Comment by cclinet** [comment]
In preview version of pytorch , the issue already exists.
```
Versions of relevant libraries:
[pip3] numpy==1.26.3
[pip3] onnx==1.15.0
[pip3] onnxruntime==1.17.0
[pip3] onnxscript==0.1.0.dev20240205
[pip3] pytorch-triton==3.0.0+901819d2b6
[pip3] torch==2.3.0.dev20240204+cu121
[pip3] torch-tensorrt==2.3.0.dev20240110+cu121
[pip3] torchvision==0.18.0.dev20240204+cu121
[conda] numpy                     1.26.3                   pypi_0    pypi
[conda] pytorch-triton            3.0.0+901819d2b6          pypi_0    pypi
[conda] torch                     2.3.0.dev20240204+cu121          pypi_0    pypi
[conda] torch-tensorrt            2.3.0.dev20240110+cu121          pypi_0    pypi
[conda] torchvision               0.18.0.dev20240204+cu121          pypi_0    pypi
```

**Comment by BowenBao** [comment]
cc @thiagocrepaldi @titaiwangms

**Comment by titaiwangms** [comment]
The repro looks correct to me? "2" is symbolic value actually.
https://github.com/onnx/onnx/blob/3d976ff359fe22c8f881a15119c3f61e497dd86a/onnx/onnx.proto#L671-L676

If you try it with dynamic_shapes=False, it becomes `dim_values`:
name: "l_x_"
type {
  tensor_type {
    elem_type: 1
    shape {
      dim {
        dim_value: 2
      }
      dim {
        dim_value: 2
      }
      dim {
        dim_value: 2
      }
    }
  }
}

**Comment by thiagocrepaldi** [comment]
> The repro looks correct to me? "2" is symbolic value actually. https://github.com/onnx/onnx/blob/3d976ff359fe22c8f881a15119c3f61e497dd86a/onnx/onnx.proto#L671-L676
> 
> If you try it with dynamic_shapes=False, it becomes `dim_value`:
> 
> ```
> name: "l_x_"
> type {
>   tensor_type {
>     elem_type: 1
>     shape {
>       dim {
>         dim_value: 2
>       }
>       dim {
>         dim_value: 2
>       }
>       dim {
>         dim_value: 2
>       }
>     }
>   }
> }
> ```

Agreed. Try running with ORT and use different dimensions for the export and then for executing the model

**Comment by cclinet** [comment]
Oh, I overlooked the quotation marks.
